content/reference-transaction-format.md

```diff
@@ -25,6 +26,63 @@ Any signature type can authorize any type of transaction, with the following exc
+
+***TODO: I feel like some concept docs on private and public keys would be great -- like when to use them. Talk about the master_key, master_seed, master_seed_hex, public_key, and public_key_hex, for example, and why we provide them and what are the use cases. It would also be good to provide concept doc on how keys are used to generate the "signature" and the different types of signature-like values that we generate: Signature, SigningPubKey, and TxnSignature, for example. What are the use cases? I believe this info is in the doc, but discussed in the context of tutorials.***
+
+### Master Keys
+
+When you create an account using the [`wallet_propose`](reference-rippled.html#wallet-propose) method, the `master_seed` (private key) and `public_key` returned in the response are the **master keys** for the account. These master keys are an intrinsic part of the account's identity as the `account_id` (account address), also returned in the `wallet_propose` response, and the `public_key` value are mathematically related.
+
+Here's a sample `wallet_propose` response, including the `master_seed` (private key), `public_key`, and `account_id` (account address) values:
+
+```
+{
+  "result": {
+    "account_id": "rM8oJwkKtgEU2HZkCHsJScLHqegWpgBRxt",
+    "key_type": "secp256k1",
+    "master_key": "SAY ARMY KITE LETS PAN WAVE JUKE ACRE MAT TILT FEAT LED",
+    "master_seed": "PrivateKeyXXXXXXXXXXXXXXXXXXX",
+    "master_seed_hex": "47E8F7F59D0CC9A3F01FD3D69CCAE93A",
+    "public_key": "aBQEK3r6hb8isnCWmortkNdv3MSMKP74o7gj3kHZQF9vDkbXo4Qs",
+    "public_key_hex": "032D911AE9887278110A8FE25169546D4E95A82AF5EFA25525FC3FDFFDC5A8B604"
+  },
+  "status": "success",
+  "type": "response"
+}
+```
+
+Master keys have more powerful signing authority than regular keys. In addition to being able to sign all transactions that regular keys can, master keys are the only keys that can be used to perform the following actions:
+
+* Use the master private key to [disable the master public key](#accountset-flags).
+* Permanently give up the ability to [freeze counterparties](concept-freeze.html#no-freeze).
+
+**Important:** Master keys can be disabled, but cannot be changed. This means that if your master private key is [compromised](concept-issuing-and-operational-addresses.html), you cannot change it to stop the compromise.
+
+This is a compelling reason to keep your master keys offline whenever possible and set up regular keys to sign transactions from your master account instead.
+
+
+### Regular Keys
+
+When you create an account using the [`wallet_propose`](reference-rippled.html#wallet-propose) method, the `master_seed` (private key) and `public_key` returned in the response are the master keys for the account. For the purposes of this discussion, let's call this account the master account. Ripple recommends that you keep the master account's master keys offline whenever possible and assign regular keys to the master account to sign your transactions instead.
```

Comment:
Let's not use "account" to refer to a key pair; that makes things more confusing. It's not an account until it's funded and exists in the ledger. The `wallet_propose` command—and other ways to generate keypairs, not built into `rippled`—generate a "key pair" (sometimes called a wallet, but that term is also overloaded, so we avoid it), which is just a pair of related numbers.

The only way to create an "account" in the XRP Ledger is for someone who has XRP already to send a Payment to a valid address. Doing so creates an account at that address. The XRP Ledger neither knows nor cares how that address was generated. The protocol doesn't even know for sure that a valid pair of keys exists for that address—it's an open (possibly unanswerable) question whether there are addresses without valid keys and and it's assumed that there are potentially multiple valid key pairs for any given address (but it's so hard to find any valid key pair for a given address that it's fine).

In case you're keeping track, the list of ways _not_ to refer to "the thing consisting of a public key, private key, and an address" includes:

- **wallet** (don't want to confuse it with "software for managing your balances")
- **account** (don't want to confuse it with "the thing that's stored in a ledger" also known as "an identifier with a transaction history and metadata")

I mean, "key pair" is not a great way to refer to it either since a key pair generally means "just a private key and a public key, not this weird construct that also has a hash of the public key tagging along OMG why is it here" but I think it's the best we've got. 🙀